scripts/globals/spells/paralyze_ii.lua

```diff
@@ -14,61 +14,54 @@ function onMagicCastingCheck(caster,target,spell)
 end;
 
 function onSpellCast(caster,target,spell)
+    local merits = caster:getMerit(MERIT_PARALYZE_II);
 
-    if (target:hasStatusEffect(dsp.effects.PARALYSIS)) then -- Effect already on, do nothing
-        spell:setMsg(msgBasic.MAGIC_NO_EFFECT);
-    else
-        -- Calculate duration.
-        local duration = 120;
+    -- Pull base stats
+    local dMND = caster:getStat(MOD_MND) - target:getStat(MOD_MND);
 
-        if (caster:hasStatusEffect(dsp.effects.SABOTEUR)) then
-            duration = duration * 2;
-        end
+    -- Base potency
+    local potency = math.floor(dMND / 4) + 20;
 
-        local dMND = caster:getStat(MOD_MND) - target:getStat(MOD_MND);
+    if (potency > 30) then
+        potency = 30;
+    end
 
-        -- Calculate potency.
-        local merits = caster:getMerit(MERIT_PARALYZE_II);
-        local potency = math.floor(dMND / 4) + 20;
-        if (potency > 30) then
-            potency = 30;
-        end
+    if (potency < 10) then
+        potency = 10;
+    end
 
-        if (potency < 10) then
-            potency = 10;
-        end
+    if (merits > 1) then
+        potency = potency + merits - 1;
+    end
 
-        if (merits > 1) then
-            potency = potency + merits - 1;
-        end
+    if (caster:hasStatusEffect(dsp.effects.SABOTEUR)) then
+        potency = potency * 2;
+    end
 
+    local duration = 120;
+    local params = {};
+    params.diff = nil;
+    params.attribute = MOD_MND;
+    params.skillType = 35;
+    params.bonus = merits*2;
+    params.effect = dsp.effects.PARALYSIS;
+    local resist = applyResistanceEffect(caster, target, spell, params);
+
+    if (duration >= 90) then
```

Comment:
still has the same inconsistency I pointed out before. Nothing is going to change it from 120 before it hits here, so this condition will always be the same. it will always be greater or equal to 90, because its already 120. It isn't until farther down after this check that you have resistance altering the duration.